File: 2023/day02.py
import re
data = open("input-2.txt","r")
d = data.readlines()
total = 0
for i in d:
    game_id = int(re.search("Game (\d+)",i)[1])
    t = i.split(';')
    tmp_bool = True
    for s in t:
        r,g,b = 0,0,0
        if re.search("(\d+) blue",s):
            b = int(re.search("(\d+) blue",s)[1])
        if re.search("(\d+) red",s):
            r = int(re.search("(\d+) red",s)[1])
        if re.search("(\d+) green",s):
            g = int(re.search("(\d+) green",s)[1])
        if b > 14 or g > 13 or r > 12:
            tmp_bool = False
            break
    if tmp_bool:
        total = total + game_id
            
print(total)    
    
power = 0
for i in d:
    t = i.split(';')
    r,g,b = 0,0,0
    for s in t:
        if re.search("(\d+) blue",s):
            b = max(b,int(re.search("(\d+) blue",s)[1]))
        if re.search("(\d+) red",s):
            r = max(r,int(re.search("(\d+) red",s)[1]))
        if re.search("(\d+) green",s):
            g = max(g,int(re.search("(\d+) green",s)[1]))
    power = power + r*g*b
    
print(power)
data.close()

File: 2023/day02.jl
data = readlines("2023/input-2.txt")
# const r, g, b = 12, 13, 14
d = Dict("r" => 12, "g" => 13, "b" => 14)
re1 = r"Game (\d+):"
re2 = r"(\d+) ([r,g,b])"
c = 0
for l in data
    game_ID = parse(Int,match(re1,l)[1])
    spl_1 = split(l,[';',':'])[2:end]
    tmp_bool = true
    for l2 in spl_1
        spl_2 = split(l2,',')
        # println(match(re2, spl_2[1]))
        tmp_bool &= all([d[match(re2,k)[2]] >= parse(Int,match(re2,k)[1]) for k in spl_2])
    end
    tmp_bool && (c += game_ID)
end
println(c)

c = 0
for l in data
    game_ID = parse(Int, match(re1, l)[1])
    spl_1 = split(l, [';', ':'])[2:end]
    tmp_bool = true
    for l2 in spl_1
        spl_2 = split(l2, ',')
        # println(match(re2, spl_2[1]))
        tmp_bool &= all([d[match(re2, k)[2]] >= parse(Int, match(re2, k)[1]) for k in spl_2])
    end
    tmp_bool && (c += game_ID)
end
println(c)